### PR TITLE
add reference to DNS over HTTPS

### PR DESCRIPTION
**Pull request by yt0ng**

### CODE REVIEW

File: nursery/reference-doh.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
+rule:
+  meta:
+    name: reference of DoH
```
Comment (williballenthin) [inline nursery/reference-doh.yml:3]: lets rename this to "references DNS over HTTPS endpoints"

File: nursery/reference-doh.yml [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    namespace: communication/dns
+    author: markus.neis@swisscom.com
+    scope: file
+    references:
+      - https://github.com/curl/curl/wiki/DNS-over-HTTPS
+    examples:
+      # - 749e7becf00fccc6dff324a83976dc0d
```
Comment (williballenthin) [inline nursery/reference-doh.yml:10]: can you update this line to be:
  1. not commented, and
  2. include the function offset in which the string is found?
    a. you can use `capa -v` to figure out what its recognizing
    b. format it like `749e7...:0x401000`

File: nursery/reference-doh.yml [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  features:
+    - or:
+      - string: /.*https://doh.seby.io:8443/dns-query.*/i
```
Comment (williballenthin) [inline nursery/reference-doh.yml:13]: based on your example, i see that we do need to use regex here to account for application data being passed in the url. 

do you think its reasonable to remove the wildcard at the start of each of the strings here? regex matching is one area that capa is not very performant, and i think having a static anchor at the start of each pattern can maybe maintain performance as we add lots of patterns. 

image

**Review by williballenthin — CHANGES_REQUESTED** [review]
i think this rule is really neat! i'm excited to get it merged. a little bit of feedback included inline. let me know if anything is unreasonable and we're happy to work with you.

File: nursery/reference-doh.yml (hunk 1 above)
Comment (williballenthin) [reply nursery/reference-doh.yml:3]: will need to rename the file, too

File: nursery/reference-doh.yml (hunk 2 above)
Comment (williballenthin) [reply nursery/reference-doh.yml:10]: and finally, would you add the example file to https://github.com/fireeye/capa-testfiles/

File: nursery/reference-doh.yml (hunk 1 above)
Comment (yt0ng) [reply nursery/reference-doh.yml]: Done!

File: nursery/reference-doh.yml (hunk 2 above)
Comment (yt0ng) [reply nursery/reference-doh.yml]: > can you update this line to be:
> 
> 1. not commented, and
> 2. include the function offset in which the string is found?
>    a. you can use `capa -v` to figure out what its recognizing
>    b. format it like `749e7...:0x401000`

Done!

Comment (yt0ng) [reply nursery/reference-doh.yml]: > and finally, would you add the example file to https://github.com/fireeye/capa-testfiles/

Added in https://github.com/fireeye/capa-testfiles/pull/32

File: nursery/reference-doh.yml (hunk 3 above)
Comment (mr-tz) [reply nursery/reference-doh.yml]: We decided to use `re.search` to make rule writing more convenient.
https://github.com/fireeye/capa/blob/d5186f160dd461eb4fa779d9dc75f0766aedbf4f/capa/features/__init__.py#L151-L155

Comment (yt0ng) [reply nursery/reference-doh.yml]: removed the wildcard at the start but apparently not necessary thanks @mr-tz

File: nursery/reference-doh.yml (hunk 2 above)
Comment (yt0ng) [reply nursery/reference-doh.yml]: Correction: had to close the previous PR and added now in https://github.com/fireeye/capa-testfiles/pull/33 adjusted filename

**Pull request merged**